oauth2_http/java/com/google/auth/oauth2/LoggingConfigs.java

```diff
@@ -0,0 +1,50 @@
+package com.google.auth.oauth2;
+
+import org.slf4j.ILoggerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class LoggingConfigs {
+
+  private static EnvironmentProvider environmentProvider = SystemEnvironmentProvider.getInstance();
+  private static final Logger NO_OP_LOGGER = org.slf4j.helpers.NOPLogger.NOP_LOGGER;
+  private static boolean loggingEnabled = isLoggingEnabled();
+  // expose this setter only for testing purposes
+  static void setEnvironmentProvider(EnvironmentProvider provider) {
+    environmentProvider = provider;
+    // Recalculate LOGGING_ENABLED after setting the new provider
+    loggingEnabled = isLoggingEnabled();
+  }
```

Comment:
I see. Having these methods as static makes this harder. Not the biggest fan of this and almost would rather us do the maven profile workaround to test this.

It's package-private and used only for testing. We can try and address this last.

Reply:
Due to time constraints, I am leaning towards keep this logic for now, and make necessary changes in the next iteration/release if possible.

That said, can you share your concerns about this?

Reply:
The main concern is just having a setter that potentially changes the state of the logging. Calling `setEnvironmentProvider()` ends up re-calculating if logging is enabled.

For now, it's package private and would only be called be us (for now it's only used for tests). I think it's fine for now but ideally I don't think it's something that we should have

Reply:
Sounds good. It's package-private and we should be able to change easily in the future if something better comes up.